src/cc_catalog_airflow/dags/provider_api_scripts/metropolitan_museum_of_art.py

```diff
@@ -0,0 +1,279 @@
+
+
+def extract_the_data(object_ids):
+    for i in object_ids:
+        get_data_for_each_image(i)
+
+
+def get_data_for_each_image(object_id):
+
+    endpoint = 'https://collectionapi.metmuseum.org/public/collection/v1/objects/{}'.format(object_id)
+
+    object_json = _get_response_json(None, endpoint)
+
+    if object_json is None:
+        logger.error('Unable to process object ID : {}'.format(object_id))
+        return None
+
+    message = object_json.get('message')
+    if message:
+        logger.warning(f'{message} : {object_id}')
+        return None
+
+    isCC0 = object_json.get('isPublicDomain')
+    if (isCC0 is None) or isCC0 is False:
+        logger.warning('CC0 license not detected')
+        return None
+
+    foreign_url = object_json.get('objectURL', None)
+    if foreign_url is None:
+        logger.warning(f'No landing page detected for: {object_id}')
+        return None
+
+    image_info = object_json.get('primaryImage', None)
+    if image_info is None:
+        logger.warning(f'No image found for {object_id}')
+        return None
+
+    title = object_json.get('title')
+    creator_name = object_json.get('artistDisplayName')
+    foreign_id = object_id
+    meta_data = create_meta_data(object_json)
+    image_url = image_info
+
+    thumbnail = ''
+    if '/origina/' in image_url:
+        thumbnail = image_url.replace('/original/', '/web-large/')
+
+    other_images = object_json.get('additionalImages', None)
+
+    if other_images is not None and len(other_images)>1:
+        extra_image_index = 1
+        meta_data['set'] = foreign_url
+
+
+    image_store.add_item(
+        foreign_url,  # foreign url of image
+        image_url,  # image url
+        thumbnail,  # thubnail url
+        None,  # license URl
+        'cc0',  # license
+        '1.0',  # license verion
+        foreign_id,  # foreign identifier
+        None,  # width
+        None,  # height
+        creator_name,  # creator name
+        None,  # creator url
+        title,  # title
+        meta_data,  # meta data
+        None,
+        'f',
+        None
+    )
+
+    for image in other_images:
+        foreign_id = '{}-{}'.format(object_id, extra_image_index)
+        image_url = image
+        thumbnail = ''
+
+        if image_url:
+            if '/original/' in image_url:
+                image_url.replace('/original/', '/web-image/')
+
+        image_store.add_item(
+            foreign_url,  # foreign url of image
+            image_url,  # image url
+            thumbnail,  # thubnail url
+            None,  # license URl
+            'cc0',  # license
+            '1.0',  # license verion
+            foreign_id,  # foreign identifier
+            None,  # width
+            None,  # height
+            creator_name,  # creator name
+            None,  # creator url
+            title,  # title
+            meta_data,  # meta data
+            None,
+            'f',
+            None
+    )
+
+
+
+def create_meta_data(object_json):
```

Comment:
Take care of the naming convention, all the non-class methods should start with an `_`. Change all the function names. e.g create_meta_data,extract_the_data

Reply:
It's actually not all non-class methods, it's 'private' methods, i.e., ones that are not intended to be imported by anything that loads the module.

Reply:
I have made the required changes though

Reply:
Oh, thanks for the correction @mathemancer .